src/feature_builder.hpp

```diff
@@ -220,49 +223,70 @@ struct overzoomed_feature_builder
     {
         std::vector<detail::annotated_ring<CoordinateType>> rings;
         vtzero::decode_polygon_geometry(feature.geometry(), detail::polygon_handler<CoordinateType>(rings, dx_, dy_, zoom_factor_));
-        if (!rings.empty())
+        if (rings.empty())
         {
-            vtzero::polygon_feature_builder feature_builder{layer_builder_};
-            feature_builder.copy_id(feature);
-            bool valid = false;
-            bool process = false;
-            for (auto& r : rings)
+            return;
+        }
+        mapbox::geometry::multi_polygon<coordinate_type> results;
+        mapbox::geometry::wagyu::wagyu<std::int64_t> wagyu;
+        bool data_added = false;
+        bool skip_inner = true;
+        for (auto const& r : rings)
+        {
+            if (r.second == vtzero::ring_type::outer)
             {
-                if (r.second == vtzero::ring_type::outer)
+                if (data_added)
                 {
-                    auto extent = mapbox::geometry::envelope(r.first);
-                    process = boost::geometry::intersects(extent, bbox_);
+                    wagyu.execute(mapbox::geometry::wagyu::clip_type_union,
```

Comment:
What's the motivation for doing each outer ring and its children separately? Does it work better than a single cleaning of the entire geometry?

Reply:
This will result in the multipolygons being unioned -- this was not desired behavior for some situations where polygon edges might touch for some users. It is not technically valid but we haven't seen render issues from this and it helps for situations such as multiple small buildings touching where a user wants to see each outline.